Geometry/MTDGeometryBuilder/src/MTDGeomUtil.cc

```diff
@@ -0,0 +1,209 @@
+#include "Geometry/MTDGeometryBuilder/interface/MTDGeometry.h"
+#include "Geometry/MTDGeometryBuilder/interface/ProxyMTDTopology.h"
+#include "Geometry/MTDGeometryBuilder/interface/RectangularMTDTopology.h"
+#include "Geometry/MTDNumberingBuilder/interface/MTDTopology.h"
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+
+using namespace mtd;
+
+void MTDGeomUtil::setGeometry(const MTDGeometry* geom) { geom_ = geom; }
+
+void MTDGeomUtil::setTopology(const MTDTopology* topo) { topology_ = topo; }
+
+bool MTDGeomUtil::isETL(const DetId& id) const {
+  MTDDetId hid{id};
+  const auto& subDet = hid.mtdSubDetector();
+  if (subDet == 0)
+    throw cms::Exception("mtd::MTDGeomUtil") << "DetId " << hid.rawId() << " not in MTD!" << std::endl;
+  if (subDet == MTDDetId::MTDType::ETL)
+    return true;
+  return false;
+}
+
+bool MTDGeomUtil::isBTL(const DetId& id) const { return !(isETL(id)); }
+
+// row and column set to 0 by default since they are not needed for BTL
+std::pair<LocalPoint, GlobalPoint> MTDGeomUtil::getPosition(const DetId& id, int row, int column) const {
+  LocalPoint local_point(0., 0., 0.);
+  GlobalPoint global_point(0., 0., 0.);
+  if (isBTL(id)) {
+    BTLDetId detId{id};
+    DetId geoId = detId.geographicalId(MTDTopologyMode::crysLayoutFromTopoMode(topology_->getMTDTopologyMode()));
+    const MTDGeomDet* thedet = geom_->idToDet(geoId);
+    if (thedet == nullptr)
+      throw cms::Exception("mtd::MTDGeomUtil") << "GeographicalID: " << std::hex << geoId.rawId() << " ("
+                                               << detId.rawId() << ") is invalid!" << std::dec << std::endl;
+    const ProxyMTDTopology& topoproxy = static_cast<const ProxyMTDTopology&>(thedet->topology());
+    const RectangularMTDTopology& topo = static_cast<const RectangularMTDTopology&>(topoproxy.specificTopology());
+
+    local_point = topo.pixelToModuleLocalPoint(local_point, detId.row(topo.nrows()), detId.column(topo.nrows()));
+    global_point = thedet->toGlobal(local_point);
+  } else if (isETL(id)) {
+    ETLDetId detId{id};
+    DetId geoId = detId.geographicalId();
+    const MTDGeomDet* thedet = geom_->idToDet(geoId);
+    if (thedet == nullptr)
+      throw cms::Exception("mtd::MTDGeomUtil") << "GeographicalID: " << std::hex << geoId.rawId() << " ("
+                                               << detId.rawId() << ") is invalid!" << std::dec << std::endl;
+    const ProxyMTDTopology& topoproxy = static_cast<const ProxyMTDTopology&>(thedet->topology());
+    const RectangularMTDTopology& topo = static_cast<const RectangularMTDTopology&>(topoproxy.specificTopology());
+
+    local_point = LocalPoint(topo.localX(row), topo.localY(column), 0.);
+    global_point = thedet->toGlobal(local_point);
+  } else {
+    throw cms::Exception("mtd::MTDGeomUtil") << "detId " << id.rawId() << " not in MTD" << std::endl;
```

Comment:
I've removed the `else` that cannot be reached (here and below). `isETL` contains a check on the `DetId` being in MTD, so that should be enough to handle any situation.